frontend/src/components/widgets/CameraInput/CameraInput.tsx

```diff
@@ -0,0 +1,579 @@
+
+      this.setState({
+        imgSrc: null,
+      })
+
+      this.props.widgetMgr.setFileUploaderStateValue(
+        this.props.element,
+        newWidgetValue,
+        { fromUi: true }
+      )
+    })
+  }
+
+  public render = (): React.ReactNode => {
+    const { element, widgetMgr, disabled, width } = this.props
+
+    // Manage our form-clear event handler.
+    this.formClearHelper.manageFormClearListener(
+      widgetMgr,
+      element.formId,
+      this.onFormCleared
+    )
+
+    if (!this.state.imgSrc) {
+      return (
+        <StyledCameraInput
+          width={width}
+          className="row-widget"
+          data-testid="stCameraInput"
+        >
+          <WidgetLabel label={element.label} disabled={disabled}>
+            {element.help && (
+              <StyledWidgetLabelHelp>
+                <TooltipIcon
+                  content={element.help}
+                  placement={Placement.TOP_RIGHT}
+                />
+              </StyledWidgetLabelHelp>
+            )}
+          </WidgetLabel>
+          <WebcamComponent
+            handleCapture={this.handleCapture}
+            width={width}
+            disabled={disabled}
+            inClearPhotoMode={this.state.inClearPhotoMode}
+            setInClearPhotoMode={this.setInClearPhotoMode}
+          />
+        </StyledCameraInput>
+      )
+    }
+
+    return (
+      <StyledCameraInput
+        width={width}
+        className="row-widget"
+        data-testid="stCameraInput"
+      >
+        <WidgetLabel label={element.label} disabled={disabled}>
+          {element.help && (
+            <StyledWidgetLabelHelp>
+              <TooltipIcon
+                content={element.help}
+                placement={Placement.TOP_RIGHT}
+              />
+            </StyledWidgetLabelHelp>
+          )}
+        </WidgetLabel>
+        <StyledBox width={width}>
+          {this.state.imgSrc !== this.RESTORED_FROM_WIDGET_STRING && (
+            <img
+              src={this.state.imgSrc}
+              alt="Snapshot"
+              style={{
+                objectFit: "contain",
+                opacity:
+                  this.state.shutter || !this.state.minShutterEffectPassed
+                    ? "50%"
+                    : "100%",
+                // this may need to use theme but getting invalid hook usage
+                borderRadius: `.25rem .25rem 0 0`,
+              }}
+              width={width}
+              height={(width * 9) / 16}
+            />
+          )}
+        </StyledBox>
+        <CameraInputButton
+          onClick={this.removeCapture}
+          progress={this.getProgress()}
+          disabled={!!this.getProgress() || disabled}
+        >
+          {this.getProgress() ? (
+            "Uploading..."
+          ) : (
+            <StyledSpan>
+              <Icon content={X} margin="0 xs 0 0" size="sm" /> Clear photo
+            </StyledSpan>
+          )}
+        </CameraInputButton>
+      </StyledCameraInput>
```

Comment:
The refactoring mentioned in my comment above could be more easily done if this were split out into a helper component